cpp/src/rolling/rolling.cu

```diff
@@ -46,14 +47,127 @@ namespace experimental {
 namespace detail {
 
 namespace { // anonymous
+/**---------------------------------------------------------------------------*
```

Comment:
Isn't this the old style doxygen block?  New style doesn't have the ------------------------  stuff.